sdl-core/src/main/scala/io/smartdatalake/workflow/action/generic/transformer/EncryptColumnsTransformer.scala

```diff
@@ -56,22 +56,39 @@ trait EncryptDecrypt {
     new SecretKeySpec(keyBytes, "AES")
   }
 
-  def encrypt(message: String): String = {
+  def encryptECB(message: String): String = {
+    val cipher: Cipher = Cipher.getInstance(ECB_ALGORITHM_STRING)
+    val secretKey = generateAesKey(keyAsBytes)
+    cipher.init(Cipher.ENCRYPT_MODE, secretKey)
+    val data = cipher.doFinal(message.getBytes())
+    Base64.getEncoder.encodeToString(data)
+  }
+
+  def decryptECB(encryptedDataString: String): String = {
```

Comment:
To apply object oriented pattern you could:
Instead of mixing GCM and ECB in the same "trait", have a class implementing GCM and another one implementing ECB, both extending a common base trait EncryptDecrypt.
Then you could choose the implementation class in EncryptColumnsTransformer.

Reply:
done.